server/server.go

```diff
@@ -490,8 +490,9 @@ func (s *Server) ShowProcessList() map[uint64]*util.ProcessInfo {
 		if atomic.LoadInt32(&client.status) == connStatusWaitShutdown {
 			continue
 		}
-		pi := client.ctx.ShowProcess()
-		rs[pi.ID] = pi
+		if pi := client.ctx.ShowProcess(); pi != nil {
```

Comment:
a bugfix?

Reply:
Yes, https://github.com/pingcap/tidb/issues/10740
But I'm not sure how to add a test case for it.

Reply:
Do release 2.1 and 3.0 have this issue?

Reply:
No, only master branch has this issue now.
Before refactoring the expensive log, this function will only be called when we use `show processlist`, the `pi` can be promised to be not null.

Reply:
But this PR needs to be cherry-picked to release-3.0, so we'd better use another PR to fix it?

Reply:
Could you add an UT about it?

Reply:
Could you add an UT about it?